src/pool/srv_pool.c

```diff
@@ -1462,6 +1472,8 @@ pool_svc_step_up_cb(struct ds_rsvc *rsvc)
 		D_FREE(map_buf);
 	if (prop != NULL)
 		daos_prop_free(prop);
+	if (rc < 0)
+		ds_pool_failed_add(svc->ps_uuid, rc);
```

Comment:
If this PS replica encounters an error, such as a -DER_NOTLEADER due to a leadership loss during read_db_for_stepping_up, then the pool_svc_failed object will remain even if this replica becomes a leader again in the future. It seems in this approach we need to remove any corresponding pool_svc_failed object when rc is 0, no?

Reply:
Good points, thanks!